api/src/ISceneRunner.cs

```diff
@@ -33,7 +34,19 @@ public interface ISceneRunner : IDisposable
     /// <param name="autofree">If true the loaded scene will be automatic freed when the runner is freed.</param>
     /// <param name="verbose">Prints detailed infos on scene simulation.</param>
     /// <returns></returns>
-    public static ISceneRunner Load(string resourcePath, bool autofree = false, bool verbose = false) => new Core.SceneRunner(resourcePath, autofree, verbose);
+    public static ISceneRunner Load(string resourcePath, bool autofree = false, bool verbose = false)
+    {
+        if (!ResourceLoader.Exists(resourcePath))
+            throw new FileNotFoundException($"GdUnitSceneRunner: Can't load scene by given resource path: '{resourcePath}'. The resource does not exists.");
+        if (!resourcePath.EndsWith(".tscn") && !resourcePath.EndsWith(".scn") && !resourcePath.StartsWith("uid://"))
+            throw new ArgumentException($"GdUnitSceneRunner: The given resource: '{resourcePath}' is not a scene.");
+
+        var currentScene = ((PackedScene)ResourceLoader.Load(resourcePath)).Instantiate();
+
+        return Load(currentScene, autofree, verbose);
+    }
+
+    public static ISceneRunner Load(Node currentScene, bool autofree = false, bool verbose = false) => new Core.SceneRunner(currentScene, autofree, verbose);
```

Comment:
Instead of Node it should be a PackedScene, the scene loader is designed to load scenes and not custom Node implementations.

Reply:
Perhaps its my inexperience with Godot but I dont see much distinction between scenes and Nodes other than the way they are instantiated. 

Would there be a problem with allowing nodes in the interface as well for increased flexibility?

Reply:
Additionally, by only allowing a PackedScene we still can't call post-instantiation initialization methods before the scene gets instantiated and added to the scene tree.

Reply:
You are right, it should be `Node`, `PackedScene` is the loaded resource type and `Node` the base class of an instance
`public static ISceneRunner Load(Node currentScene, bool autofree = false, bool verbose = false)`